dojo/unittests/scans/nexpose/no_vuln.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<NexposeReport version="1.0">
+<scans>
+<scan id="1" name="Localhost" startTime="20131127T050242739" endTime="20131127T115513233" status="stopped"/>
+</scans><nodes>
+<node address="192.168.1.1" status="alive" device-id="7">
+<fingerprints>
+<os  certainty="0.70" device-class="General" vendor="Linux" family="Linux" product="Linux" version="2.6.9"/>
+</fingerprints>
+<tests>
+<test id="generic-icmp-netmask" status="not-vulnerable">
+
+<Paragraph>
+	<Paragraph>No response</Paragraph></Paragraph>
+</test>
+</tests>
+</service>
+</services>
+</endpoint>
```

Comment:
As the error says your xml is invalid at line 19 as there is no <endpoint> open tag.

Reply:
fix it complete waiting for TravisCI's result

Reply:
======================================================================

ERROR: test_nexpose_parser_has_many_finding (dojo.unittests.test_nexpose_parser.TestNexposeParser)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/app/dojo/unittests/test_nexpose_parser.py", line 19, in test_nexpose_parser_has_many_finding

    parser = NexposeFullXmlParser(testfile, Test())

  File "/app/dojo/tools/nexpose/parser.py", line 41, in __init__

    self.items = self.get_items(tree, vulns, test)

  File "/app/dojo/tools/nexpose/parser.py", line 297, in get_items

    Endpoint(host=item['name'], product=test.engagement.product))

  File "/usr/local/lib/python3.6/site-packages/django/db/models/fields/related_descriptors.py", line 189, in __get__

    "%s has no %s." % (self.field.model.__name__, self.field.name)

dojo.models.Test.engagement.RelatedObjectDoesNotExist: Test has no engagement.

======================================================================

ERROR: test_nexpose_parser_has_no_finding (dojo.unittests.test_nexpose_parser.TestNexposeParser)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/app/dojo/unittests/test_nexpose_parser.py", line 9, in test_nexpose_parser_has_no_finding

    parser = NexposeFullXmlParser(testfile, Test())

  File "/app/dojo/tools/nexpose/parser.py", line 40, in __init__

    vulns = self.get_vuln_definitions(tree)

  File "/app/dojo/tools/nexpose/parser.py", line 160, in get_vuln_definitions

    vid = vulnDef.get('id').lower()

AttributeError: 'NoneType' object has no attribute 'lower'

======================================================================

ERROR: test_nexpose_parser_has_one_finding (dojo.unittests.test_nexpose_parser.TestNexposeParser)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/app/dojo/unittests/test_nexpose_parser.py", line 15, in test_nexpose_parser_has_one_finding

    self.assertEqual(1, len(self.parser.items))

AttributeError: 'TestNexposeParser' object has no attribute 'parser'

----------------------------------------------------------------------
Here you go another problem lol. i will remove self from parser and try it again

Reply:
And again another error 
======================================================================

ERROR: test_nexpose_parser_has_many_finding (dojo.unittests.test_nexpose_parser.TestNexposeParser)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/app/dojo/unittests/test_nexpose_parser.py", line 19, in test_nexpose_parser_has_many_finding

    parser = NexposeFullXmlParser(testfile, Test())

  File "/app/dojo/tools/nexpose/parser.py", line 41, in __init__

    self.items = self.get_items(tree, vulns, test)

  File "/app/dojo/tools/nexpose/parser.py", line 297, in get_items

    Endpoint(host=item['name'], product=test.engagement.product))

  File "/usr/local/lib/python3.6/site-packages/django/db/models/fields/related_descriptors.py", line 189, in __get__

    "%s has no %s." % (self.field.model.__name__, self.field.name)

dojo.models.Test.engagement.RelatedObjectDoesNotExist: Test has no engagement.

======================================================================

ERROR: test_nexpose_parser_has_no_finding (dojo.unittests.test_nexpose_parser.TestNexposeParser)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/app/dojo/unittests/test_nexpose_parser.py", line 9, in test_nexpose_parser_has_no_finding

    parser = NexposeFullXmlParser(testfile, Test())

  File "/app/dojo/tools/nexpose/parser.py", line 40, in __init__

    vulns = self.get_vuln_definitions(tree)

  File "/app/dojo/tools/nexpose/parser.py", line 160, in get_vuln_definitions

    vid = vulnDef.get('id').lower()

AttributeError: 'NoneType' object has no attribute 'lower'

======================================================================

FAIL: test_nexpose_parser_has_one_finding (dojo.unittests.test_nexpose_parser.TestNexposeParser)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/app/dojo/unittests/test_nexpose_parser.py", line 15, in test_nexpose_parser_has_one_finding

    self.assertEqual(1, len(parser.items))

AssertionError: 1 != 0

Reply:
Final Error Guys

======================================================================

ERROR: test_nexpose_parser_has_many_finding (dojo.unittests.test_nexpose_parser.TestNexposeParser)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/app/dojo/unittests/test_nexpose_parser.py", line 20, in test_nexpose_parser_has_many_finding

    parser = NexposeFullXmlParser(testfile, Test())

  File "/app/dojo/tools/nexpose/parser.py", line 41, in __init__

    self.items = self.get_items(tree, vulns, test)

  File "/app/dojo/tools/nexpose/parser.py", line 297, in get_items

    Endpoint(host=item['name'], product=test.engagement.product))

  File "/usr/local/lib/python3.6/site-packages/django/db/models/fields/related_descriptors.py", line 189, in __get__

    "%s has no %s." % (self.field.model.__name__, self.field.name)

dojo.models.Test.engagement.RelatedObjectDoesNotExist: Test has no engagement.

Please help me to find out this problem. Thanks you